packages/ui/src/proposals/components/AddProposalButton.tsx

```diff
@@ -1,37 +1,46 @@
-import React, { useCallback } from 'react'
+import React, { useCallback, useMemo } from 'react'
 
 import { useApi } from '@/api/hooks/useApi'
 import { TransactionButton } from '@/common/components/buttons/TransactionButton'
 import { PlusIcon } from '@/common/components/icons/PlusIcon'
+import { Tooltip } from '@/common/components/Tooltip'
 import { useFirstObservableValue } from '@/common/hooks/useFirstObservableValue'
 import { useModal } from '@/common/hooks/useModal'
 import { AddNewProposalModalCall } from '@/proposals/modals/AddNewProposal'
 
 export const AddProposalButton = () => {
+  const { api } = useApi()
   const { showModal } = useModal()
-  const addNewProposalModal = useCallback(() => {
-    showModal<AddNewProposalModalCall>({
-      modal: 'AddNewProposalModal',
-    })
-  }, [])
+  const addProposalModal = useCallback(() => showModal<AddNewProposalModalCall>({ modal: 'AddNewProposalModal' }), [])
 
-  const { api } = useApi()
-  const maxProposals = api?.consts.proposalsEngine.maxActiveProposalLimit
-  const currentProposals = useFirstObservableValue(
-    () => api?.query.proposalsEngine.activeProposalCount(),
-    [api?.isConnected]
-  )
-  const areProposalSlotsAvailable = api && maxProposals && currentProposals?.lt(maxProposals)
+  const maxProposals = useMemo(() => api?.consts.proposalsEngine.maxActiveProposalLimit, [api?.isConnected])
+  const proposals = useFirstObservableValue(() => api?.query.proposalsEngine.activeProposalCount(), [api?.isConnected])
+  const availableSlots = useMemo(() => Number(maxProposals) - Number(proposals), [proposals, maxProposals])
+
+  const tooltipProps = useMemo(() => {
+    if (!api?.isConnected) return { tooltipText: 'Connecting to api' }
+    if (!api?.isConnected || availableSlots > 0) {
+      return {
+        tooltipText: 'Use the proposal engine to suggest a change to the Council.',
+        tooltipLinkText: 'Learn about the Proposal System',
+        tooltipLinkURL: 'https://joystream.gitbook.io/testnet-workspace/system/proposal-system',
+      }
+    } else {
+      return {
+        tooltipTitle: 'Max active proposals limit reached',
+        tooltipText: `The creation of proposals is currently disabled because the number of deciding or gracing proposals reached the limit of ${maxProposals}.`,
+        tooltipLinkText: 'Proposal System Constants',
+        tooltipLinkURL: 'https://joystream.gitbook.io/testnet-workspace/system/proposal-system#constants',
+      }
+    }
+  }, [api?.isConnected, availableSlots])
 
   return (
-    <TransactionButton
-      style="primary"
-      size="medium"
-      onClick={addNewProposalModal}
-      disabled={!areProposalSlotsAvailable}
-    >
-      <PlusIcon />
-      Add new proposal
-    </TransactionButton>
+    <Tooltip placement="bottom-end" {...tooltipProps}>
+      <TransactionButton style="primary" size="medium" onClick={addProposalModal} disabled={!availableSlots}>
```

Comment:
`TransactionButton` also renders a [Tooltip when a transaction is pending](https://github.com/Joystream/pioneer/blob/6fb1e843fefeb4bf3a72674417821d0fab25e254/packages/ui/src/common/components/buttons/TransactionButton.tsx#L18). As a result in case of a pending transaction, this will result in the Tooltip beeing nested (I'm not sure what will actually be shown). To avoid this situation, `TransactionButton` could accept `TooltipContentProp`:
- If these are defined ⇒ display the Tooltip
- If a transaction is pending ⇒ display the: "Please wait until the current transaction is over" regardless of whether a different tooltip text was passed.

WDYT ?

Reply:
This makes sense.